src/telemetry.c

```diff
@@ -977,17 +977,22 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
         size_t payload_len;
         int ret = 0;
 
-        payload_len = strlen((char *)payload);
-
-        if (payload_len > MAX_PAYLOAD_LENGTH) {
-                return -EINVAL;
+        if (payload == NULL) {
+                telem_log(LOG_WARNING, "payload pointer is NULL\n");
+                return -ENOMEM;
```

Comment:
Considering this doesn't allocate I'm not sure why -ENOMEM is returned but I'm not familiar with the codebase.